dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/BedrockModelUtilities.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Amazon.BedrockRuntime;
+using Amazon.BedrockRuntime.Model;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.Amazon.Core;
+
+/// <summary>
+/// Utilities class for functions all Bedrock models need to use.
+/// </summary>
+internal static class BedrockModelUtilities
+{
+    /// <summary>
+    /// Maps the AuthorRole to the corresponding ConversationRole because AuthorRole is static and { readonly get; }. Only called if AuthorRole is User or Assistant (System set outside/beforehand).
+    /// </summary>
+    /// <param name="role"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    internal static ConversationRole MapAuthorRoleToConversationRole(AuthorRole role)
+    {
+        if (role == AuthorRole.User)
+        {
+            return ConversationRole.User;
+        }
+
+        if (role == AuthorRole.Assistant)
+        {
+            return ConversationRole.Assistant;
+        }
+        throw new ArgumentException($"Invalid role: {role}");
```

Comment:
Can be simplified.

```suggestion
        return role switch
        {
            AuthorRole.User => ConversationRole.User,
            AuthorRole.Assistant => ConversationRole.Assistant,
            _ => throw new ArgumentException($"Invalid role: {role}")
        };
```

Reply:
A constant value is expected for AuthorRoles (only has `{ readonly get; }` attribute) so changing to `=>` would cause build error

Reply:
You are right, totally forgot we have a struct. 👍🏻